### PR TITLE
Replace <iron-dropdown>

### LINKED ISSUE #40: Improve iOS fullscreen UX
See if there are ways to enhance the end user experience, compared to the feature/fullscreen-ios branch prototype, including:

- Double tapping on elements outside the dropdown list
- Touch dragging other elements than the dropdown list
- Refocusing the input field

### LINKED ISSUE #72: Overlay alignment and scrolling lock breaks on Safari when pinch zooming
Tried on Safari 8 and 9, after doing some pinch zooming, overlay is misplaced and the page can be scrolled.

Related with polymerelements/iron-dropdown#39

### LINKED ISSUE #74: Using the “tab” buttons on iOS when the fullscreen mode doesn’t work as expected
The two blue arrow keys above the software keyboard on iOS work unexpectedly.

When you have the fullscreen mode open for a combo box, and press either of the “tab” buttons to move the prev/next field, the combo box overlay stays open.

Similary, when “tabbing” into a combo box on a “fullscreen device”, should the overlay open automatically. Most likely not, as it might cause a too big interruption if the user is just “tabbing through”, but this needs some discussion.

### LINKED ISSUE #78: vaadin-combo-box focus causes page width to change in certain browsers
If vaadin-combo-box gets focus, it hides page scroll bar reducing page width on browsers that have non-overlaid scroll bar (e.g. Chrome on Windows and Chromium on Ubuntu, not OS X). Change to page width will align centered content again making the layout jump on each focus and blur.

To test, open https://cdn.vaadin.com/vaadin-elements/master/vaadin-combo-box/demo/ in non-OS X browser and focus any combo box.

### LINKED ISSUE #80: Too many events are fired when opening for the first time
When a combo-box is opened for the first time, it fires at least 3 `opened-changed` events and 2 `iron-overlay-opened` events. 

This is most probably because the `<vaadin-adaptive-overlay>` doesn't have a default value for `_verticalAlign` and it closes and reopens the overlay while trying to figure the alignment out.

### LINKED ISSUE #81: Add screenshot of fullscreen mode to readme
It would be great to promote the responsive behavior of the element in the screenshot.

The screenshot should probably include the element running inside an actual device.

### LINKED ISSUE #84: Fullscreen layout breaks down on Android
In the UX test proto the layout broke down on Android 5 test device. The error occurred at least when there was more than one instance of a combo box with the same items inside a form ("Languages" combo box in the proto).
- Options list was cut half from the middle even if there was so many items that they didn't fit in the viewport.
- Some options appeared as empty rows.

Background: [UX Test Notes](https://docs.google.com/document/d/1h2OkslUkD3m_QG3VXJS-QOCrEgci6qCiqDPT7MPzaKo/edit#bookmark=id.qk1y70nfcc9m)

### LINKED ISSUE #86: Closing fullscreen mode causes zoom level issues on iOS
After closing the fullscreen mode, iOS sometimes zooms in the page instead of returning to the original view. In some tests it was even more frequently reproduced if the HTML document already has a viewport meta tag.

`<meta name="viewport" content="width=device-width">`

From http://wc.demo.vaadin.com/combo-box-usertest/index.html:
screen shot 2015-12-11 at 10 36 33

### LINKED ISSUE #92: combo-box incompatible with paper-header-panel?
Opening a combo-box on a page with a `paper-header-panel`, shows it incorrectly grayed out and under the header on iPhone. 

screenshot_20151214-090812

img_0347

### LINKED ISSUE #93: Don't close overlay when clicking on the input field
Currently, when trying to refocus the input field, overlay is closed, hence resetting the value typed.

This is especially noticeable in a touch device running desktop mode.

Instead, we should just focus the input field and closing should be done either by clicking outside the overlay or clicking the button.

### LINKED ISSUE #98: Enable scrolling when overlay is open
`<iron-dropdown>` disables scrolling by default when the overlay is open. 

If the body would be scrollable instead, it would allow user to scroll hidden items into view more easily.

Update: decided to get rid of `<iron-dropdown>` altogether to:
 - Enable scrolling by default and remove the need for a fullscreen mode and vertical alignment
 - Use `position: absolute` for the overlay to mitigate iOS Safari issues
 - Place the overlay directly under body to handle `z-index` and scrolling containers

### LINKED ISSUE #101: Page scrollbar disappears and content shifts to the right after clicking inside combo box
With Chrome 47 on Ubuntu, clicking inside the input field of the combo box on https://cdn.vaadin.com/vaadin-elements/master/vaadin-combo-box/demo/ causes the page scrollbar to disappear and shifts the page content to the right.  I'm guessing this also happens on Windows but not on OS X where the scrollbar is hidden?

### PR DESCRIPTION
**Pull request by Saulis**
**Note for reviewers:** due to the magnitude of the changes, it might be easier to first look through the new implementations by themselves and then comparing the test files with the old ones to see how the features have changed.

Main changes:
- `<iron-dropdown>` and it's dependencies dropped
- Fullscreen mode dropped
- Automatic vertical alignment dropped
- Overlay is placed directly under `document.body`
- Overlay uses `position: absolute``
- Scrolling is enabled, overlay position is updated automatically.

Closed issues: (list being updated)
- Fixes #98
- Fixes #92
- Fixes #93
- Fixes #86
- Fixes #84
- Fixes #40
- Closes #81
- Fixes #80
- Fixes #78
- Fixes #101
- Fixes #74
- Fixes #72

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-combo-box/105)

### CODE REVIEW

**Comment by tomivirkki** [comment]
Review status: 0 of 17 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-dropdown-behavior.html, line 14 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84720PiiJ9WzxqI7ff:-K84720PiiJ9WzxqI7fg:-640950703) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L14)):*
Is there a reason to extract these three references as properties. Seems to be working just as well with "this.$.input" instead of "this._focusTarget"

---

*[vaadin-overlay-behavior.html, line 31 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K847vHtN-NkiS19NHCw:-K847vHtN-NkiS19NHCx:621930142) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-overlay-behavior.html#L31)):*
This should check if the overlay gets attached to body and only listen in that case. Now _setOverlayPosition is fired for on every scroll even if the vaadin-combo-box is closed

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tehapo** [comment]
Review status: 0 of 17 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-combo-box.html, line 157 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K848cNpf0oDoOxHG8kR:-K848cNpf0oDoOxHG8kS:-1449445324) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box.html#L157)):*
Looks like this observer is not even defined as a function?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tehapo** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-combo-box-overlay.html, line 7 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K849FCFjeNGrExIFZji:-K849FCGCkMzmF_i15cn:669400584) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box-overlay.html#L7)):*
Import for <iron-list> is missing.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tehapo** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-combo-box-overlay.html, line 7 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K849FCFjeNGrExIFZji:-K849r4zq3c05KtbyMNB:563630657) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box-overlay.html#L7)):*
Import for `iron-list` is missing. Looks like adding the actual tag here in the comment did not display correctly. :smile: 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by Saulis** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-combo-box-overlay.html, line 7 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K849FCFjeNGrExIFZji:-K84Mj60OqZJtgRnHS0M:-928641720) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box-overlay.html#L7)):*
:+1:  will fix.

---

*[vaadin-combo-box.html, line 157 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K848cNpf0oDoOxHG8kR:-K84Ma4vZdzf9X4rRAMa:-506873981) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box.html#L157)):*
You mean `close` ? It's defined in the `vaadin-dropdown-behavior.html`

---

*[vaadin-dropdown-behavior.html, line 14 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84720PiiJ9WzxqI7ff:-K84LsptO9FK0F17WbIg:1704561739) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L14)):*
Well the original idea I had was to make this more generic by making it easier to override those properties. But now that I think of it, it probably doesn't make sense at this point.

I'll remove the properties.

---

*[vaadin-overlay-behavior.html, line 31 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K847vHtN-NkiS19NHCw:-K84MPPmkmQTrInbdLgP:730160692) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-overlay-behavior.html#L31)):*
Yes that would make  sense.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tehapo** [comment]
Review status: 0 of 17 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-combo-box.html, line 157 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K848cNpf0oDoOxHG8kR:-K84NKw0PYhC6X3rQTSI:-1472118591) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box.html#L157)):*
Oh, right. Nevermind then. :smile: 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by Saulis** [comment]
Review status: 0 of 18 files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[vaadin-combo-box-overlay.html, line 7 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K849FCFjeNGrExIFZji:-K84NWXTDKrH9aZ9GE8d:-499146009) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-combo-box-overlay.html#L7)):*
Done.

---

*[vaadin-dropdown-behavior.html, line 14 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84720PiiJ9WzxqI7ff:-K84OpVzJLSbD8GIzPP9:-499146009) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L14)):*
Done.

---

*[vaadin-overlay-behavior.html, line 31 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K847vHtN-NkiS19NHCw:-K84PKONsm1b-z8j2ONV:-499146009) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-overlay-behavior.html#L31)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tomivirkki** [comment]
*[vaadin-dropdown-behavior.html, line 2 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84QWzRc4yQPZcm97mM:-K84QWzRc4yQPZcm97mN:-1567115853) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L2)):*
The behaviors should be namespaced under vaadin.elements.combobox (for example) like the stuff in vaadin-grid. Currently it's window.VaadinDropdownBehavior

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by Saulis** [comment]
Review status: 0 of 18 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-dropdown-behavior.html, line 2 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84QWzRc4yQPZcm97mM:-K84SCkREdJhlrdYWS0V:-2122987120) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L2)):*
Polymer is using `window.Polymer`, should we just use `Vaadin.DropdownBehavior` to follow the same convention? IMO `vaadin.elements.combobox` is quite long and Java-like :vhs: 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tomivirkki** [comment]
*[vaadin-dropdown-behavior.html, line 2 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84QWzRc4yQPZcm97mM:-K84T4vObrbwQpn6-GN_:1664275003) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L2)):*
Agreed. vaadin.elements.combobox is unnecessarily long but that's what vaadin-grid uses currenlty. It would be weird to have the other one use window.Vaadin and the other window.vaadin.elements

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by Saulis** [comment]
Review status: 0 of 18 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-dropdown-behavior.html, line 2 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84QWzRc4yQPZcm97mM:-K84UwVVnWHeTDgh76qr:2085084040) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L2)):*
`vaadin.elements.DropdownBehavior` then?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tomivirkki** [comment]
*[vaadin-dropdown-behavior.html, line 2 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84QWzRc4yQPZcm97mM:-K84Vb1y2JWqHjT4CKZC:-1531833134) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L2)):*
Sounds about right

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by Saulis** [comment]
Review status: 0 of 18 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-dropdown-behavior.html, line 2 \[r7\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K84QWzRc4yQPZcm97mM:-K84Ww5mRy4XAR0YDonx:-499146009) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/b7ffc241caf15eb9a02862f7b935979224e0b264/vaadin-dropdown-behavior.html#L2)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by manolo** [comment]
In android when clicking on the input to focus it, the input is covered by the keyboard. Shouldn't we move the input to the visible view port.
Another issue is that randomly when focusing the input the overlay does not open. Unable to find a pattern tough.

---

Reviewed 11 of 17 files at r1, 1 of 3 files at r4, 2 of 6 files at r8, 3 of 4 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[test/toggling-dropdown.html, line 25 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K85y64ZB9ZgyHHO7lqT:-K85y64ZB9ZgyHHO7lqU:-854714663) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/test/toggling-dropdown.html#L25)):*
isn't clearer & shorter a regexp :  `/^(?!chrome).*safari/i.test(navigator.userAgent)` 
Android former user agent was something like  `...Android .... Safari ...` but I guess we don't want to support it anyway

---

*[test/toggling-dropdown.html, line 35 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K860S8DsHURsKMO_Mtd:-K860S8DsHURsKMO_Mte:370818534) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/test/toggling-dropdown.html#L35)):*
Doesn't work Polymer.Base.fire ?

---

*[vaadin-combo-box-overlay.html, line 23 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K861GoBCipi-x0uj3U3:-K861GoCXDM0YgVB-Yes:280809011) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-combo-box-overlay.html#L23)):*
was the same problem than we saw in android ? 

---

*[vaadin-combo-box-overlay.html, line 173 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K864HfnlXGncP_iDJ3p:-K864HfnlXGncP_iDJ3q:1384903749) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-combo-box-overlay.html#L173)):*
- In my chrome seems not working, and it's not possible to move parent's scroll at all.
- This approach does not work in touch devices that does not have the `wheel` event
- I think the best approach would be to always prevent default and don't pass scrolling to parent at all in all devices. To me is upsetting to move paren scrolling in menus. It should simplify the code as well not having to execute this calculations in all movements.

---

*[vaadin-dropdown-behavior.html, line 18 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K868j68hclE8md6e9XA:-K868j68hclE8md6e9XB:709174317) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-dropdown-behavior.html#L18)):*
We should copy this JS doc to the combo box so as gwt-api-generator could pick them up. I have to double check though

By the way if we navigate to the api doc, clicking on behaviours links does not open the doc.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-:-K867q3qoFNmi2DcN8VH:1603496619)*

**Comment by tomivirkki** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-overlay-behavior.html, line 7 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K8IjK7v3kf4JJRse4_I:-K8IjK7v3kf4JJRse4_J:1918885477) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-overlay-behavior.html#L7)):*
Since this object isn't used anywhere directly there's no need to expose it.

vaadin.elements.combobox.OverlayBehavior = [{....}, Polymer.IronResizableBehavior]

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by platosha** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[vaadin-overlay-behavior.html, line 7 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K8IjK7v3kf4JJRse4_I:-K8IkU_9O-KtvUbhEJ1d:-533660506) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-overlay-behavior.html#L7)):*
Right, but:

1. That's the way Polymer documentation suggests. Iron* behaviors are also written this way.

2. Having a definition inside an array requires additional indentation level.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by Saulis** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[test/toggling-dropdown.html, line 25 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K85y64ZB9ZgyHHO7lqT:-K8IiiquUqfnbNcyTCRJ:-610074043) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/test/toggling-dropdown.html#L25)):*
Well, as a non-regexp master like myself, I find the current version more understandable :smiley_cat: 

---

*[test/toggling-dropdown.html, line 35 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K860S8DsHURsKMO_Mtd:-K8IkDw_aYD5cX53fe6g:1022356800) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/test/toggling-dropdown.html#L35)):*
Yes it actually does! :+1: 

---

*[vaadin-combo-box-overlay.html, line 23 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K861GoBCipi-x0uj3U3:-K8IkK0Vu5wt_5vzNBEA:-505039693) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-combo-box-overlay.html#L23)):*
Well kinda, in Safari, the scroll bars were getting hidden by item background-color when hovering, on Android the scrollbars were getting over the whole overlay.

---

*[vaadin-combo-box-overlay.html, line 173 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K864HfnlXGncP_iDJ3p:-K8IklowsOFFvbPMkwez:-1796396737) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-combo-box-overlay.html#L173)):*
> In my chrome seems not working, and it's not possible to move parent's scroll at all.

That's how it's designed to work, see the "Futher improvement idea" in the function comments. I would do any possible improvements in a separate PR.

> This approach does not work in touch devices that does not have the wheel event

This patch is only targeted for desktops, since in touch devices the scrolling already works as expected (parent is scrolled after list is scrolled to start/end).

> I think the best approach would be to always prevent default and don't pass scrolling to parent at all in all devices.

I don't think we want to do that, it would be painfully difficult to scroll the body on a touch device after that. Now it works like every other scrollable content.

---

*[vaadin-dropdown-behavior.html, line 18 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K868j68hclE8md6e9XA:-K8Io0TCQBvol1K4B4_z:-1685342530) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-dropdown-behavior.html#L18)):*
Should the generator pick up the events from the inherited behaviors?

The behaviors were missing `/** @polymerBehavior */` annotations, added them.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by manolo** [comment]
LGTM

---

Reviewed 4 of 4 files at r12.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[test/toggling-dropdown.html, line 25 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K85y64ZB9ZgyHHO7lqT:-K8JhCeV5vGKU0EOCIL6:1901559975) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/test/toggling-dropdown.html#L25)):*
What about shorter code? this is a test though.

---

*[vaadin-combo-box-overlay.html, line 23 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K861GoBCipi-x0uj3U3:-K8Jj6glorid7P1QAs_3:1951688904) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-combo-box-overlay.html#L23)):*
So could you extend this comment mentioning the android issue as well ?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-:-K8JjYmJZmIQAxjCSZPs:676834233)*

File: vaadin-combo-box-overlay.html [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      position: absolute;
+      z-index: 100;
+      @apply(--shadow-elevation-2dp);
+      background: #fff;
+      border-radius: 0 0 2px 2px;
+    }
+
+    #scroller {
+      overflow: auto;
+      max-height: 65vh;
```
Comment (jounik) [inline vaadin-combo-box-overlay.html]: This value only works okay if `<meta name="viewport" content="width=device-width, initial-scale=1" />` is used. On a larger viewport size the dropdown will also be larger, and when for instance iOS zooms in on the focused input, the dropdown will not fit inside the zoomed viewport.

Doesn’t block this PR, but something to consider in the future, if we can somehow improve the “guesstimate” of the dropdown height.

**Comment by jounik** [comment]
#70 could be closed together with this, but I would still like to re-evaluate the possibility to use some other alignment for the dropdown than just below. Probably only on desktop and in some specific cases, but still, it might be needed in the future.

I’ll update the ticket to include "above" alignment also, and we can come back to it in the future.

**Comment by Saulis** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[vaadin-combo-box-overlay.html, line 23 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K861GoBCipi-x0uj3U3:-K8TmY8Yog5E8TgMRGdA:-1573261257) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-combo-box-overlay.html#L23)):*
Well, there was a separate fix for Android which had comments, but Tomi actually removed the fix after changing the overlay position calculations to use translate in commit d46c818. 

---

*[vaadin-overlay-behavior.html, line 7 \[r11\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-K8IjK7v3kf4JJRse4_I:-K8TmyISjleXbt8i1Alv:-677551349) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/3e6dd55726d3061c6723b3185add090ccd13bf59/vaadin-overlay-behavior.html#L7)):*
Yeah, I would do it the same way Polymer does their behaviours.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105)*

**Comment by tomivirkki** [comment]
LGTM

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-:-K8ULD4mACuGjjL9Ye6r:676834233)*

**Comment by tehapo** [comment]
:shipit: :rocket: :lgtm_strong: :dash: 

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/105#-:-K8ULLrOVTUrQKOKNzJi:-1158353678)*

**Pull request merged**